torch/testing/_comparison.py

```diff
@@ -275,11 +275,20 @@ def __init__(
         self._unknown_parameters = unknown_parameters
 
     @staticmethod
-    def _check_inputs_isinstance(*inputs: Any, cls: Union[Type, Tuple[Type, ...]]):
-        """Checks if all inputs are instances of a given class and raise :class:`UnsupportedInputs` otherwise."""
+    def _check_inputs_isinstance(
+        *inputs: Any, cls: Union[Type, Tuple[Type, ...]], not_cls: Optional[Union[Type, Tuple[Type, ...]]] = None
+    ) -> None:
+        """Checks if all inputs are instances of a given class and and optionally not of another.
```

Comment:
An interesting extension (that might be more logically separated into its own function)